Python solution for week2

hienhpss opened the pull request.

vietlq commented on week0002/hienhpss/python/week0002.py:
```diff
@@ -0,0 +1,37 @@
+def spi_round(n=4,m=4,x=0):
+	'''Looping through the spiral path to return the next
+	cell'''
+	if (x>=(n+1)/2) or (x>=(m+1)/2):
```
Spacing please! Give some breathing space for operators

vietlq commented:
Please remove the *.pyc file too. It's not part of submission.

vietlq commented on week0002/hienhpss/python/week0002.py:
```diff
@@ -0,0 +1,37 @@
+	for i in range(m-x-1,x,-1):
+		yield(n-x-1,i)
+	for i in range(n-x-1,x,-1):
+		yield(i,x)
```
Avoid generators for now :) Why complicating the issue when you can just use loops? 🍡

vietlq commented on week0002/hienhpss/python/week0002.py:
```diff
@@ -0,0 +1,37 @@
+def spiral(n=4,m=4):
+	'''Return the n*m board with numbers
+	arrange in spiral format'''
+	board = [[False for x in range(m)] for y in range(n)]
```
Also: `m*[n*[False]]`

hienhpss commented:
Fixed the spacing issue and removed pyc file. The generator thing is a personal choice towards writing more pythonic code (and it works better for bigger problem compare with initializing a huge list).

vietlq commented on week0002/hienhpss/python/week0002.py:
```diff
@@ -0,0 +1,37 @@
+	cell'''
+	if (x >= (n+1)/2) or (x >= (m+1)/2):
+		return
+	for i in range(x,m-x-1):
```
Spacing please. All operators: `+ - * / % == , =` should have space around them, except `=` in function argument passing.

hienhpss replied:
Done

vietlq commented on week0002/hienhpss/python/week0002.py:
```diff
@@ -0,0 +1,37 @@
+		for next_row,next_col in spi_round(n,m,i):
+			if board[next_row][next_col] == False:
+				board[next_row][next_col] = count
+				count+=1
```
Space around `+= -= /= *=` too

hienhpss replied:
Done

hienhpss commented on week0002/hienhpss/python/week0002.py:
```diff
@@ -0,0 +1,37 @@
+def spi_round(n=4,m=4,x=0):
```
Done

vietlq commented on week0002/hienhpss/python/week0002.py:
```diff
@@ -0,0 +1,37 @@
+	count = 1
+	for i in range(limit):
+		for next_row, next_col in spi_round(n, m, i):
+			if board[next_row][next_col] == False:
```
More Pythonic way to do it: `if not board[next_row][next_col]:`

vietlq commented on week0002/hienhpss/python/week0002.py:
```diff
@@ -0,0 +1,37 @@
+	for i in range(x, m-x-1):
+		yield (x, i)
+	for i in range(x, n-x-1):
+		yield(i, m-x-1)
```
Spacing around `-` too.

hienhpss replied:
Done
